### PR TITLE
RedisVectorStoreDriver Updates

### PR DESCRIPTION
**Pull request by trish11953**

### CODE REVIEW

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 1]
```diff
@@ -90,20 +91,47 @@ def query(
...
+        for document, vector in zip(results, vectors):
             query_results.append(
                 BaseVectorStoreDriver.QueryResult(
-                    vector=vector_float_list,
+                    vector=np.frombuffer(vector, dtype=np.float32).tolist(),
                     score=float(document['score']),
-                    meta=metadata,
-                    namespace=namespace
+                    meta=None,
+                    namespace=None
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:107]: Why did you change those to `None`?

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 2]
```diff
@@ -90,20 +91,47 @@ def query(
 
         results = self.client.ft(self.index).search(query_expression, query_params).docs
 
-        query_results = []
+        pipe = self.client.pipeline()
         for document in results:
-            metadata = getattr(document, "metadata", None)
-            namespace = document.id.split(":")[0] if ":" in document.id else None
-            vector_float_list = json.loads(document["vec_string"])
+            pipe.hget(document.id, "vector")
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:96]: Looks like this line executes an extra request to Redis for every result. Not sure we want that. Previous implementation didn't have that problem.

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 3]
```diff
@@ -90,20 +91,47 @@ def query(
...
+        except:
+            schema = (
+                TagField("tag"),
+                VectorField("vector",
+                            "FLAT", {
+                                "TYPE": "FLOAT32",
+                                "DIM": self.vector_dimensions,
+                                "DISTANCE_METRIC": "COSINE",
+                            }
+                            ),
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:125]: Formatting should look like this:

```
VectorField(
    "vector",
    "FLAT", {
        "TYPE": "FLOAT32",
        "DIM": self.vector_dimensions,
        "DISTANCE_METRIC": "COSINE",
    }
)
```

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 4]
```diff
@@ -90,20 +91,47 @@ def query(
...
+                    meta=None,
+                    namespace=None
                 )
             )
         return query_results
 
+    def create_index(self, namespace: Optional[str] = None):
+        try:
+            self.client.ft(self.index).info()
+            print("Index already exists!")
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:115]: Use logging instead.

**Review by vasinov — CHANGES_REQUESTED** [review]

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 5]
```diff
@@ -15,6 +19,7 @@ class RedisVectorStoreDriver(BaseVectorStoreDriver):
     db: int = field(kw_only=True, default=0)
     password: Optional[str] = field(default=None, kw_only=True)
     index: str = field(kw_only=True)
+    vector_dimensions: int = field(kw_only=True, default=1536)
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:22]: Let's move this into `create_index` parameters, since it's only used there. Let's also not have a default, since embedding dimensions vary from model to model.

File: griptape/drivers/vector/redis_vector_store_driver.py [hunk 6]
```diff
@@ -105,5 +110,29 @@ def query(
             )
         return query_results
 
+    def create_index(self, namespace: Optional[str] = None):
+        try:
+            self.client.ft(self.index).info()
+            logging.info("Index already exists!")
```
Comment (vasinov) [inline griptape/drivers/vector/redis_vector_store_driver.py:116]: Should it be an error?

**Review by vasinov — CHANGES_REQUESTED** [review]

**Review by vasinov — COMMENTED** [review]
Just realized that `query()` takes a vector instead of a string. Should be a string, so it's consistent with other drivers and the abstract method in `BaseVectorStoreDriver`.

File: griptape/drivers/vector/redis_vector_store_driver.py (hunk 6 above)
Comment (trish11953) [reply griptape/drivers/vector/redis_vector_store_driver.py:116]: It shouldn't be an error since it's okay if the index exists right?

Comment (vasinov) [reply griptape/drivers/vector/redis_vector_store_driver.py:116]: Yeah, may be not an error then but a `warning`, since the method technically failed to create a new index.

Also, could you add ` -> None` to the signature, since we are not returning anything here?

Comment (trish11953) [reply griptape/drivers/vector/redis_vector_store_driver.py]: Done!

**Review by vasinov — APPROVED** [review]
